ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java

```diff
@@ -70,15 +71,15 @@
   private static final String DEFAULT_REPLICATION_PROP = "default.replication.factor";
   private static final String DELETE_TOPIC_ENABLE = "delete.topic.enable";
 
-  private final Admin adminClient;
+  private final MemoizedSupplier<Admin> adminClientSupplier;
 
   /**
    * Construct a topic client from an existing admin client.
    *
-   * @param adminClient the admin client.
+   * @param adminClientSupplier the admin client supplier.
    */
-  public KafkaTopicClientImpl(final Admin adminClient) {
-    this.adminClient = Objects.requireNonNull(adminClient, "adminClient");
+  public KafkaTopicClientImpl(final MemoizedSupplier<Admin> adminClientSupplier) {
```

Comment:
There's no reason for this class to _know_ about `MemoizedSupplier`.  Rather than have it accept a `MemoizedSupplier`, just instantiate this class lazily as well. When you need to instantiate it, first instantiate the `Admin` client.